src/components/ProductList.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useState, useEffect } from 'react'
+import ProductListItem from './ProductListItem'
+import { Container, LinearProgress } from '@material-ui/core'
+import Product from '../domain/Product';
+import { getAll } from '../data/ProductService';
+
+
+function ProductList() {
+    const initialProducts = [] as Product[]
+    const [products, setProducts] = useState(initialProducts)
+    const [isLoading, setIsLoading] = useState(true)
+
+    useEffect(() => {
+        getAll()
+        .then(result => {
+            setProducts(result.data)
+            setIsLoading(false)
+        })
```

Comment:
You can add a callback for the error case here as well, as you did in _deleteProduct_ from _ProductDetails_ component